src/components/Navbar.tsx

```diff
@@ -1,11 +1,53 @@
 import { Navbar as HospitalRunNavbar } from '@hospitalrun/components'
 import React from 'react'
 import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
 import { useHistory } from 'react-router-dom'
 
+import Permissions from '../model/Permissions'
+import { RootState } from '../store'
+
 const Navbar = () => {
+  const { permissions } = useSelector((state: RootState) => state.user)
   const { t } = useTranslation()
   const history = useHistory()
+
+  const addPages = [
+    {
+      permission: Permissions.WritePatients,
+      label: t('patients.newPatient'),
+      path: '/patients/new',
+    },
+    {
+      permission: Permissions.WriteAppointments,
+      label: t('scheduling.appointments.new'),
+      path: '/appointments/new',
+    },
+    {
+      permission: Permissions.RequestLab,
+      label: t('labs.requests.new'),
+      path: '/labs/new',
+    },
+    {
+      permission: Permissions.ReportIncident,
+      label: t('incidents.reports.new'),
+      path: '/incidents/new',
+    },
+  ]
+
+  const addDropdownList: { type: string; label: string; onClick: () => void }[] = []
+  addPages.forEach((page) => {
+    if (permissions.includes(page.permission)) {
+      addDropdownList.push({
+        type: 'link',
+        label: page.label,
+        onClick: () => {
+          history.push(page.path)
+        },
+      })
+    }
+  })
```

Comment:
Feel free to ignore this, but a more functional of writing this could be. I think it may be more readable, but I could be in the minority here. 

```ts
  const addDropdownList: { type: string; label: string; onClick: () => void }[] = addPages
    .filter((page) => permissions.includes(page.permission))
    .map((page) => ({
      type: 'link',
      label: page.label,
      onClick: () => {
        history.push(page.path)
      },
    }))
```

Reply:
such a great idea! ✅ in cd9e4bd